common/src/main/java/net/william278/huskclaims/hook/DatabaseImporter.java

```diff
@@ -0,0 +1,310 @@
+            
+            // Backup SQLite database file
+            Path sourceFile = getPlugin().getConfigDirectory().resolve(SQLITE_DB_NAME);
+            Path targetFile = backupsDir.resolve("HuskClaimsData_" + timestamp + ".db");
+            Files.copy(sourceFile, targetFile);
+            getPlugin().log(Level.INFO, "Created SQLite database backup at " + targetFile);
+        } catch (Exception e) {
+            getPlugin().log(Level.WARNING, "Failed to create database backup: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Sets up the source and target databases for migration
+     *
+     * @throws IllegalStateException if database initialization fails
+     */
+    private void setupDatabases() {
+        // Get source database (current database)
+        sourceDatabase = getPlugin().getDatabase();
+
+        // Create target database based on target type
+        if (targetType.equalsIgnoreCase("mysql")) {
+            targetDatabase = new MySqlDatabase(getPlugin());
+        } else {
+            targetDatabase = new SqLiteDatabase(getPlugin());
+        }
+        
+        // Initialize target database
+        targetDatabase.initialize();
+        if (!targetDatabase.hasLoaded()) {
+            throw new IllegalStateException("Failed to initialize target database");
+        }
+    }
+
+    /**
+     * Reset statistical counters for a new migration
+     */
+    private void resetCounters() {
+        userCount = 0;
+        groupCount = 0;
+        claimWorldCount = 0;
+    }
+
+    @Override
+    protected int importData(@NotNull ImportData importData, @NotNull CommandUser executor) {
+        return switch (importData) {
+            case USERS -> importUserData(executor);
+            case CLAIMS -> importClaimData(executor);
+            default -> 0;
+        };
+    }
+
+    /**
+     * Import user data including saved users and user groups
+     *
+     * @param executor The command user executing the import
+     * @return The number of users imported
+     */
+    private int importUserData(CommandUser executor) {
+        try {
+            // Get and import all users (including inactive)
+            allUsers = sourceDatabase.getInactiveUsers(-1);
+            
+            // Import users
+            for (SavedUser user : allUsers) {
+                targetDatabase.createOrUpdateUser(user);
+                userCount++;
+            }
+            
+            // Import user groups - doing it here ensures we have all users first
+            allUserGroups = sourceDatabase.getAllUserGroups();
+            for (Map.Entry<UUID, Set<UserGroup>> entry : allUserGroups.entrySet()) {
+                for (UserGroup group : entry.getValue()) {
+                    targetDatabase.addUserGroup(group);
+                    groupCount++;
+                }
+            }
+            
+            return userCount;
+        } catch (Exception e) {
```

Comment:
I prefer catching Throwable instead of exception as some exception classes don't extend Exception (as throwable is actually the base class)